remote_version.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"context"
+
+	"github.com/hansmi/paperhooks/pkg/client"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type remoteVersionClient interface {
+	GetRemoteVersion(ctx context.Context) (*client.RemoteVersion, *client.Response, error)
+}
+
+type remoteVersionCollector struct {
+	cl remoteVersionClient
+
+	updateAvailableDesc *prometheus.Desc
+}
+
+func newRemoteVersionCollector(cl remoteVersionClient) *remoteVersionCollector {
+	return &remoteVersionCollector{
+		cl: cl,
+
+		updateAvailableDesc: prometheus.NewDesc("paperless_remote_version_update_available", "Whether an update is available.", []string{"version"}, nil),
+	}
+}
+
+func (c *remoteVersionCollector) describe(ch chan<- *prometheus.Desc) {
+	ch <- c.updateAvailableDesc
+}
+
+func (c *remoteVersionCollector) collect(ctx context.Context, ch chan<- prometheus.Metric) error {
+	remoteVersion, _, err := c.cl.GetRemoteVersion(ctx)
+	if err != nil {
+		return err
+	}
+
+	if len(remoteVersion.Version) == 0 {
+		return nil
+	}
```

Comment:
Shouldn't the metric still be reported, just with an empty version string?

Reply:
It can. I would say it is a design decision.
From the "human" context, I ask myself: If we don't have a version string, how useful would it be to know if an update is available? I mean, what are you updating?

Reply:
A lack of a version string could be reported as an error (see separate comments on failures). Let's leave it for this PR.

Reply:
I created https://github.com/hansmi/prometheus-paperless-exporter/issues/66